sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -285,4 +289,52 @@ export class EducationProgramOfferingController {
       studyStartDate: offering.studyStartDate,
     };
   }
+
+  /**
+   * Get programs for a particular institution in paginated.
+   * @param institutionId id of the institution.
+   * @param take is the number of rows shown in the table
+   * @param skip is the number of rows that is skipped/offset from the total list.
+   * For example page 2 the skip would be 10 when we select 10 rows per page.
+   * @param dateSubmittedOrder the sorting order of the submitted date, default its ascending
+   * @param searchProgramName Search the program name in the query
+   * @returns ProgramsOfferingSummaryPaginated.
+   */
+  @AllowAuthorizedParty(AuthorizedParties.aest)
+  @Groups(UserGroups.AESTUser)
+  @Get("institution/:institutionId/programs/take/:take/skip/:skip")
+  async getPaginatedProgramsForInstitution(
+    @Param("institutionId") institutionId: number,
+    @Param("take") take: number,
```

Comment:
Should take and skip not be also query string parameters?

Reply:
I thought of it and decided to go for the Param, as this will give us difference from the paginated and not paginated routes.

Reply:
maybe you can give something like `@Get("institution/:institutionId/programs/query`, to avoid route conflict and get take and skip it as query parameter? thoughts @dev

Reply:
For paginated routes, I see there is always a take and skip irrespective of the page loading, I feel it would be better to have it as a Param, rather than in the query. As query params are more of optional parameters and params are like always present.